src/goSignature.ts

```diff
@@ -43,7 +43,7 @@ export class GoSignatureHelpProvider implements SignatureHelpProvider {
 				let funcName = declarationText.substring(0, nameEnd);
 				sig = declarationText.substring(sigStart);
 				si = new SignatureInformation(funcName + sig, res.doc);
-			} else {
+			} else if (res.toolUsed === 'gogetdoc') {
```

Comment:
Latest changes should fix this - if guru is set as the docsTool, I just passed in a config object with "godoc" set as the docsTool for provideHover and provideSignatureHelp.